src/include/OSL/oslexec.h

```diff
@@ -254,7 +254,8 @@ class OSLEXECPUBLIC ShadingSystem
         return attribute (name, TypeDesc::FLOAT, &f);
     }
     bool attribute (string_view name, string_view val) {
-        const char *s = val.c_str();
+        std::string valstr(val);
```

Comment:
Will this create an extra copy for the local variable valstr?

Reply:
Yeah, it'll create a temporary std::string. That seems a little icky, but remember that what this is replacing was potentially creating a ustring, so it's probably not a lot worse.

This is called maybe a few dozen times, tops, to set up the shading system. So I don't think there is a legit perf issue here either way.

If this were called VERY frequently, I would add specializations for std::string and ustring that would avoid the temp string creation, but I'm not sure it's worth it for a method that's called a handful of times total.

Reply:
The same pattern happens a few times, but I think the concern is that the `void*` flavor of `attribute()` (which accepts a `TypeDesc` to say what the `void*` is -- assumes a C-style null-terminated string.

So unfortunately we can't support these cases without forcing a copy from the `string_view`. In the future we could redefine that case to accept a pointer to a `string_view`, but it would be a behavior change.

On a sufficiently modern c++ standard library, short strings would at least not go through the heap.

Reply:
Ok, so string_view (OIIO or C++17) doesn't require null terminated string.  So I guess the copy is necessary to ensure null termination.  Does make me think that the next call down might be better to add direct support for string_view (vs. const char *) assuming it's going to make its own copy anyway.

Reply:
I could make additional overloads for ustring and/or std::string that could avoid the copy. 

But these are not inner loop calls with performance implications. We're talking about at most tens of calls per execution of the program.

Reply:
For the record, current there are 11 documented ShadingSystem::attribute() settings whose value is a single string. Five of them are only useful for my debugging of the implementation of OSL.

This is not a performance issue.

Reply:
I agree this isn't a performance issue.  Worst remotely reasonable case is if you want to change all the renderer_outputs or the shadersearchpath per shader, and even then you're only going to get in the thousands of calls, maybe up to megabytes of allocations in the most extreme situations -- and in those situations that should be dwarfed by shader compile and execute time and space.